scripts/compute_request_limits.py

```diff
@@ -0,0 +1,363 @@
+                lower_bound = middle
+            else:
+                upper_bound = middle - 1
+            pbar.update(1)
+
+    # Just in case the number of tokens does not match the number of words, check number fo tokens with tokenizer
+    max_prompt_length = get_number_of_tokens(
+        prefix + " ".join(["hello"] * (lower_bound - nb_tokens_prefix - nb_tokens_suffix)) + suffix,
+        tokenizer_client,
+        tokenizer_name,
+    )
+    return {
+        "max_prompt_length": max_prompt_length,
+        "nb_tokens_prefix": nb_tokens_prefix,
+        "nb_tokens_suffix": nb_tokens_suffix,
+        "usable_max_prompt_length": max_prompt_length - nb_tokens_prefix - nb_tokens_suffix,
+    }
+
+
+def figure_out_max_prompt_length_plus_tokens(
+    model_name: str,
+    tokenizer_name: str,
+    max_prompt_length: int,
+    prefix: Optional[str] = None,
+    suffix: Optional[str] = None,
+) -> int:
+    client, tokenizer_client = get_clients(tokenizer_name)
+    if prefix is None:
+        prefix = ""
+    if suffix is None:
+        suffix = ""
+    lower_bound = 1
+    upper_bound = 2 * max_prompt_length + 1
+
+    # Check if there is a limit (some model accept as many tokens as you want)
+    if try_request(
+        client,
+        model_name,
+        tokenizer_name,
+        tokenizer_client,
+        max_prompt_length,
+        2**31 - 2 - max_prompt_length,
+        prefix,
+        suffix,
+    ):
+        print("The model has no limit on the number of tokens")
+        return -1
+    else:
+        print("The model has a limit on the number of tokens")
+
+    # Perform a dichotomy search to find the max tokens betzeen lower_bound and upper_bound
+    with tqdm(total=int(math.log2(upper_bound - lower_bound))) as pbar:
+        while lower_bound < upper_bound:
+            middle = math.ceil((lower_bound + upper_bound) / 2)
+            if try_request(
+                client, model_name, tokenizer_name, tokenizer_client, max_prompt_length, middle, prefix, suffix
+            ):
+                lower_bound = middle
+            else:
+                upper_bound = middle - 1
+            pbar.update(1)
+
+    return lower_bound + max_prompt_length
+
+
+def check_limits(
+    model_name: str,
+    tokenizer_name: str,
+    infos: Dict[str, int],
+    prefix: Optional[str] = None,
+    suffix: Optional[str] = None,
+) -> bool:
+    client, tokenizer_client = get_clients(tokenizer_name)
+    if prefix is None:
+        prefix = ""
+    if suffix is None:
+        suffix = ""
+    result: bool = True
+
+    # Check the max_prompt_length
+    if "max_prompt_length" not in infos:
+        raise ValueError("infos should contain max_prompt_length")
+    max_prompt_length = infos["max_prompt_length"]
+    if max_prompt_length < 0:
+        print("No limit on the number of tokens")
+        if not try_request(client, model_name, tokenizer_name, tokenizer_client, 2**32 - 2, 1, prefix, suffix):
```

Comment:
Shouldn't `max_tokens` be zero here, if we're only checking sequence length?

Reply:
Right